docs/proposals/fua-restrict-feature/fua-restrict-feature.md

```diff
@@ -0,0 +1,127 @@
+
+FUA-Restrict indicates to SessionD that the user's access MUST be restricted
+according to IP filters definitions or IDs provided in AVPs. This is enabled
+by sending CCA-I/U with a Final-Unit-Action AVP set to RESTRICT\_ACCESS.
+
+
+## Propostion
+
+FUA-Restrict can be enabled by including either the Restriction-Filter-Rule
+AVP or the Filter-Id AVP in the Credit-Control-Answer message.
+
+```
+  [Final-Unit-Indication]
+    {Final-Unit-Action}
+    *[Restriction-Filter-Rule]
+    *[Filter-Id]
+    [Redirect-Server]
+      {Redirect-Address-Type}
+      {Redirect-Server-Address}
+```
+
+Restriction-Filter-Rule: of type IPFilterRule (FlowDescription). This will
+need to be translated to a PolicyRule.
+Filter-Id: Provides IP packet filter identifiers (In our case, It will be a
+list of PolicyRule identifiers)
+
+It is important to note that QoS can be enabled only when using Filter-Id AVP.
+Restriction-Filter-Rule AVP would only provide allowed traffic filters.
+
+This implementation will not provide Restriction-Filter-Rule as Filter-id
+provides support of all the required use-cases. Nevertheless, 
+Restriction-Filter-Rule could be translated to restriction dynamic rules with
+lower priority and apply them similar to static rules. 
+LocalEnforcer will ensure priority in this case.
+
+As we would like to use any static rule when restriction is enabled, there will
+no change to PolicyRules definition.
+
+## How We Will Change Magma
+
+**PolicyRule**
+
+Policy Object definition remains the same as the current definition. At the
+opposite of a policy rule used in the FUA-Redirect case, there is no need to
+introduce any additional field for FUA-Restrict.
+We will be able to apply any static rule when restricting access.
+
+
+**Session Proxy**
+
+We will limit this implementation to Filter-Id AVP only as It covers the desired
+use-cases. The AVP will reference a PolicyRule Identifier.
+
+```
+  [Final-Unit-Indication]
+    {Final-Unit-Action}
+    *[Filter-Id]
+    [Redirect-Server]
+      {Redirect-Address-Type}
+      {Redirect-Server-Address}
+```
+
+The support of Restriction-Filter-Rule can be added later if required. In this
+case, ReceivedCredits will need to be updated to include a FlowDescription list.
+
+We will need to translation IPFilterRule to FlowDescription as well.
+
+
+**SessionD**
+
+ServiceAction will need to be updated to support RESTRICT\_ACCESS action. Mostly,
+we will need to be able to store a restricted list of rule ids received from
+OCS/SessionProxy.
+
+In addition, we will need to store the list of rules ID to be used when restriction
+is enabled. A method get\_restricted\_rule\_ids will be added to retrive this list
+prior to sending rules activation to pipelined.
+
+We will re-use the same pipelined client method add\_gy\_final\_action\_flow as It
+provides support for this case as well.
+
+Static Rules will be sent using their original priority and will be applied as gy
+final action flow.
+
```

Comment:
Meaning we will not do any priority manipulation in SessionD for static rules? We can leave the static rule definitions as is. And we are not attempting to resolve the `Restriction-Filter-Rule` (dynamic rule) in this change? (Sorry, if this is already specified in the doc. Just trying to make sure I'm comprehending it correctly :) )

Reply:
Yes, exactly. The first implementation will support Filter-Id only as It covers all our requirements. Thus, we will use the static rules with their predefined priorities without changes.